sklearn/gaussian_process/gpc.py

```diff
@@ -0,0 +1,180 @@
+"""Gaussian processes classification based on the Laplace approximation. """
+
+# Authors: Jan Hendrik Metzen <jhm@informatik.uni-bremen.de>
+#
+# License: BSD 3 clause
+
+import numpy as np
+from scipy.linalg import cholesky, cho_solve, solve
+from scipy.optimize import fmin_l_bfgs_b
+from scipy.special import erf
+
+from sklearn.base import BaseEstimator
+
+
+class GaussianProcessClassification(BaseEstimator):
+    """ Gaussian process classification (GPC).
+
+    The implementation is based on Algorithm 3.1, 3.2, and 5.1 of
+    ``Gaussian Processes for Machine Learning'' (GPML) by Rasmussen and
+    Williams.
+
+    Internally, the Laplace approximation is used for approximating the
+    non-Gaussian posterior by a Gaussian.
+
+    Currently, the implementation is restricted to
+      * using the logistic link function
+      * binary classification
+    """
+
+    def __init__(self, kernel, jitter=0.0):
+        self.kernel = kernel
+        self.jitter = jitter
+
+        # Values required for approximating the logistic sigmoid by
+        # error functions. coefs are obtained via:
+        # x = np.array([0, 0.6, 2, 3.5, 4.5, np.inf])
+        # b = logistic(x)
+        # A = (erf(np.dot(x, self.lambdas)) + 1) / 2
+        # coefs = lstsq(A, b)[0]
+        self.lambdas = np.array([0.41, 0.4, 0.37, 0.44, 0.39])[:, None]
+        self.coefs = np.array([-1854.8214151, 3516.89893646, 221.29346712,
+                               128.12323805, -2010.49422654])[:, None]
+
+    def fit(self, X, y):
+        # XXX: Assert that y is binary and labels are {0, 1}
+        self.X_fit_ = np.asarray(X)
```

Comment:
Since we are dealing with kernel methods, `X_fit_` (maybe --> `X_train` ?) is persisted in the model in order to be able to create the kernel prediction at test time (we need `k(X_train, X_test)`). I am worried about changes in this array between train and test time. Should there maybe be an option `copy_X_train=False` in the `__init__` which one could set to `True` if one wanted to be sure of this?